onnxscript/evaluator.py

```diff
@@ -118,6 +176,8 @@ def call_ort(schema, args, kwargs, implicit_args=None):
         opset_imports=[opset_id],
         ir_version=irbuilder.select_ir_version(schema.since_version, domain=schema.domain),
     )
+    model = onnx.shape_inference.infer_shapes(model)
```

Comment:
`model` is changed here and then serialized in `_cache_` with `SerializeToString`. The information about shape inference makes the model bigger and longer to serialize. Would it be faster to keep the original `model`? I assume this line was changed to do more thorough verifications.

Reply:
Well, actually onnxruntime session creation was failing for some test-case (because the output-type was not specified in the model). I am using the inference to get output-types here. (We could potentially avoid this in the future by calling onnx to do node-level inference, but currently that has limited functionality and Jacky is working on an ONNX PR to make it complete.)

Reply:
I would probably move the shape_inference in function `_cache_` then, and call it only if onnxruntime fails to load the model. The original model could still be used as a key for the cache.

Reply:
I think the caching can be improved and may be made more efficient. But let's do that separately, it's not the focus here. We need to select the right key for caching, but am not convinced this shape-inference will make a difference. I think the types do have to be part of the caching key, so it shouldn't make a difference.